BrainPortal/app/controllers/bourreaux_controller.rb

```diff
@@ -622,14 +622,13 @@ def file_copy #:nodoc:
     # Check if the user has access to the bourreau and the data provider
     if !bourreau.can_be_accessed_by?(current_user) || !data_provider.can_be_accessed_by?(current_user)
       render :json => { :error => "Access denied" }, :status => :forbidden
+      return
     end
 
-    # Filter out userfiles that are not accessible by the user
-    userfile_ids = userfile_ids.select do |userfile_id|
-      userfile = Userfile.find(userfile_id)
-      userfile.can_be_accessed_by?(current_user)
-    end
-
+    # Filter out userfile_ids that are not readable by the user
+    userfile_ids = Userfile.find_all_accessible_by_user(current_user, :access_requested => :read)
+                        .where(:id => userfile_ids).pluck(:id)
+    
     bourreau.send_command_copy_files(userfile_ids, data_provider_id, current_user.id)
     render :json => { :status => "ok" }
```

Comment:
While you are at it, when not return some useful information in the JSON reply, like the number of files that were actually selected for the move? (`userfile_ids.size`) ?